cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs

```diff
@@ -1069,14 +1047,89 @@ updateTemplate (SystemStart start)
     unLovelace :: Integral a => Lovelace -> a
     unLovelace (Lovelace coin) = fromIntegral coin
 
+updateCreateStakedOutputTemplate
+    :: SystemStart
+    -- Genesis delegation (not stake-based):
+    -> Map (Hash GenesisKey) (Hash GenesisDelegateKey, Hash VrfKey)
+    -- Non-delegated initial UTxO spec:
+    -> Maybe Lovelace
+    -> Int
+    -> [AddressInEra ShelleyEra]
+    -- Genesis staking: pools/delegation map & delegated initial UTxO spec:
+    -> [(Ledger.KeyHash 'Ledger.StakePool StandardCrypto, Ledger.PoolParams StandardCrypto)]
+    -> [(Ledger.KeyHash 'Ledger.Staking StandardCrypto, Ledger.KeyHash 'Ledger.StakePool StandardCrypto)]
+    -> Lovelace
```

Comment:
Can we add haddocks for these parameters?

Reply:
I added the haddocks